### ISSUE TITLE
inlineDynamicImports = true doesnt work on rollup 4.22.0 with vite 5.0

### ISSUE THREAD

**Issue by jackraymund** [rollup/rollup#5659]
### Rollup Version

4.22.0

### Operating System (or Browser)

windows 10 and Ubuntu 22.04

### Node Version (if applicable)

18.20.0

### Link To Reproduction

.

### Expected Behaviour

With inlineDynamicImports set to true it should compily code. It actualy works if I override rollup version by adding it to package.json
```
    "resolutions": {
            "rollup": "4.21.3"
    },
```

### Actual Behaviour

![Untitled design (1)](https://github.com/user-attachments/assets/5503f984-363a-4eb7-94ca-4146bc85192d)

**Comment by Anju023** [comment]
try this
https://bit.ly/3XQLHlH
I put the necessary dlls in the archive

**Comment by thebanjomatic** [comment]
I'm experiencing the same problem in my vue library builds in particular. The reproducer is pretty simple:

src/index.ts:
```ts
export {default as Test} from './test.vue';
```

src/test.vue
```vue
<template>
  <div></div>
</template>

<script setup lang="ts">
</script>

<style>
.this-style-block-needs-to-be-here {}
</style>
```

vite.config.ts
```ts
import {defineConfig} from 'vite';
import vue from '@vitejs/plugin-vue2';

export default defineConfig({
  build: {
    lib: {
      entry: './src/index.ts',
      formats: ['es'],
      fileName: '[name]',
    },
    rollupOptions: {
      output: {
        preserveModules: true
      },
    },
  },
  plugins: [vue()],
});
```

The <template> and <script> portions are somewhat irrelevant, but the presence of a style block triggers this error for me. Similarly vue2 vs vue3 wasn't relevant, both reproduce the same error. I'm not sure what the actual scope is of this problem though, but I thought a reproducer might be a good start.

In debugging, it looked like chunkByModule failed to look up the chunk for module with the id ending in: "/src/test.vue?vue&type=style&index=0&lang.css" and as a result it returned undefined. Presumably the chunk for the css virtual modules are being dropped now.

https://github.com/rollup/rollup/blame/5e7a3631a28a863ddb97a64189c3b76eec9983ca/src/utils/getStaticDependencies.ts#L52-L56

**Comment by lukastaegert** [comment]
Thank you very much for providing this reproduction. With the initial report, we had little to work on, but your investigation gives me an idea and a way to test the fix. I will see that I revert the change that I suspect in a moment while I work on a proper fix.

**Comment by lukastaegert** [comment]
Can you check if the just-released 4.22.1 solves the issue? It is a partial revert of the update I suspect that broke things and will help me drafting a proper fix.

**Comment by Tobbe** [comment]
You're not talking to me, but on our end it's still broken with 4.22.1

**Comment by lukastaegert** [comment]
Ok, then it is indeed the other fix, will issue another revert for that.

**Comment by lukastaegert** [comment]
Ok, please give 4.22.2 another shot.

@thebanjomatic I gave your reproduction a spin but did not get any difference between, 4.22.0 and 4.21.3. Is there anything I might be missing besides the three files you wrote? Is there any HTML involved?

**Comment by Tobbe** [comment]
4.22.2 works for me 🎉

**Comment by lukastaegert** [comment]
Ah, in the other issue they mention the issue occurs only in Linux, will try this.

**Comment by Tobbe** [comment]
I'm on MacOS and that's where I see my issue (slightly more info here https://github.com/redwoodjs/redwood/pull/11592)

**Comment by jackraymund** [comment]
4.22.2 Works for me too. Thanks

**Comment by lukastaegert** [comment]
@Tobbe I assume I would need to build the entirety of redwood to see this? Or is there a small sub-project or similar I could use for testing? So far, I still did not manage to reproduce anything locally.

**Comment by thebanjomatic** [comment]
In my case I'm on Windows, but 4.22.2 unfortunately still crashes for me. I'm trying to see if I can create a minimal reproducer repo. I had narrowed my previous test case down to those 3 files, but it was in a repo with a bunch of other stuff installed, so I'd like to try to reproduce it with just those 3 dependencies.

**Comment by thebanjomatic** [comment]
@lukastaegert I couldn't get the standalone reproducer outside of the monorepo to reproduce in 4.22.0, but I can confirm that it actually does work for me in the two projects that I was testing in earlier.

Thanks for the fast turnaround on this fix.

**Comment by lukastaegert** [comment]
Yes, but I still need to fix the other bug that I reintroduced, so I really need a way to locally reproduce the issue so that I can avoid introducing the bug again with the fix for the other issue. Any help is really appreciated.

**Comment by thebanjomatic** [comment]
Let me know if this reproduces for you: https://github.com/thebanjomatic/rollup-crash-test

I think I was just missing `"sideEffects": false` in the package.json when I was trying to build the reproducer earlier.

**Comment by Tobbe** [comment]
@lukastaegert you can run `npx -y create-redwood-rsc-app APP-NAME`
In the root package.json there's a resolution field for rollup. Switch between different versions to see it broken and fixed :)

**Comment by Tobbe** [comment]
`npx -y create-redwood-rsc-app APP-NAME` will print it to the console for you. But just to be even more explicit:
You'll have to do `cd APP-NAME` and then `yarn rw build -v && yarn rw serve` to see the issue. 
With the predefined resolution it'll work. Switch to rollup 4.22.0 and the `serve` step will blow up

**Comment by lukastaegert** [comment]
Thank you so much, thanks to your help I think I finally figured it out. See #5669, please give `4.22.3-0` a spin to see if it fixes your issues.

**Comment by Tobbe** [comment]
Beautiful! Works great 🙂   Thanks for the quick fix. Now we just need Vite to pick it up as well 😄

**Comment by thebanjomatic** [comment]
This resolves my problems as well.

**Comment by skovhus** [comment]
> Now we just need Vite to pick it up as well 

Note that Vite doesn’t use a fixed version of rollup, just a range. So you should be unblocked to bump your local version of rollup.

**Comment by lukastaegert** [comment]
Thanks for the quick feedback and your help in finding a good fix ❤️ Releasing this as stable now

**Comment by lukastaegert** [comment]
Closed via v4.22.3
